modoboa/admin/migrations/0015_rename_view_permissions.py

```diff
@@ -0,0 +1,44 @@
+# Generated by Django 2.1.15 on 2020-01-13 09:22
+
+import sys
+
+from django.db import IntegrityError, migrations, transaction
+
+VIEW_PERMISSION_CODENAMES = {
+    'view_aliases': 'view_alias',
+    'view_domaliases': 'view_domainalias',
+    'view_mailboxes': 'view_mailbox',
+}
+
+
+def rename_view_permissions(apps, schema_editor):
+    Permission = apps.get_model('auth', 'Permission')
+    db_alias = schema_editor.connection.alias
+    # Iterate over permissions and try to rename the codename
+    for perm in Permission.objects.using(db_alias).filter(
+        codename__in=VIEW_PERMISSION_CODENAMES.keys()
+    ):
+        codename = perm.codename
+        try:
+            with transaction.atomic():
```

Comment:
Is it really necessary? I thought migrations were already run within a transaction context...

Reply:
Yes, it is as soon as an `IntegrityError` is encountered to allow the other operations to be committed. Otherwise, a `TransactionManagementError` will be raised - see this [doc](https://docs.djangoproject.com/en/stable/topics/db/transactions/#django.db.transaction.atomic). I have also discovered that when I tried to see if it was working as excepted.

Reply:
Ok, good to know.